src/channels/views/ChannelCreate/ChannelCreate.tsx

```diff
@@ -23,24 +24,28 @@ export const ChannelCreateView = ({}) => {
 
   const handleBack = () => navigate(channelsListUrl());
 
-  const onSubmit = (data: ChannelCreate) => {
-    if (!data.channelCreate.errors.length) {
-      notify({
-        status: "success",
-        text: intl.formatMessage(commonMessages.savedChanges)
-      });
-      navigate(channelPath(data.channelCreate.channel.id));
-    }
-  };
-
   const [createChannel, createChannelOpts] = useChannelCreateMutation({
-    onCompleted: onSubmit
+    onCompleted: ({ channelCreate: { errors, channel } }: ChannelCreate) => {
+      notify(getDefaultNotifierSuccessErrorData(errors, intl));
+
+      if (!errors.length) {
+        navigate(channelPath(channel.id));
+      }
+    }
   });
 
-  const handleSubmit = (data: ChannelCreateInput) =>
+  const handleSubmit = ({
+    shippingZonesIdsToAdd,
+    currencyCode,
+    ...rest
+  }: FormData) =>
     createChannel({
       variables: {
-        input: { ...data, currencyCode: data.currencyCode.toUpperCase() }
+        input: {
+          ...omit(rest, "shippingZonesIdsToRemove"),
```

Comment:
Instead of using the `omit` helper fn use the destructured args for that:

```
const handleSubmit = ({
    shippingZonesIdsToAdd,
    currencyCode,
    shippingZonesIdsToRemove,
    ...rest
  }: FormData) =>
```

```suggestion
          ...rest,
```